pkg/resolution/resolver/git/resolver_test.go

```diff
@@ -191,7 +190,43 @@ func TestResolveNotEnabled(t *testing.T) {
 	}
 }
 
+type params struct {
+	url        string
+	revision   string
+	pathInRepo string
+	org        string
+	repo       string
+}
+
 func TestResolve(t *testing.T) {
+	// lcoal repo set up for anonymous cloning
+	// ----
+	commits := []commitForRepo{{
+		Dir:      "foo/",
+		Filename: "old",
+		Content:  "old content in test branch",
+		Branch:   "test-branch",
+	}, {
+		Dir:      "foo/",
+		Filename: "new",
+		Content:  "new content in test branch",
+		Branch:   "test-branch",
+	}, {
+		Dir:      "./",
+		Filename: "released",
+		Content:  "released content in main branch and in tag v1",
+		Tag:      "v1",
+	}}
+
+	repoPath := createTestRepo(t, commits)
+
+	// NOTE: these 3 commit shas are static because of the static content, commit msg, author info and time.
+	firstCommitSHA := "24f5ed4a93a11148ef3ebd2bda567413585b5c55"
+	secondCommitSHA := "117c5a12426a0ec9780953244e89c1e219d2e9e9"
+	thirdCommitSHA := "c618acf88328cdd2e6f5282a96529065d59ab288"
```

Comment:
nit: `commits := []string{`?

Reply:
SGTM. Also changed to dynamic commits that were returned from createRestRepo as we discussed offline. PTAL. tyty!